src/ClientMessage.ts

```diff
@@ -1,206 +1,189 @@
     }
 
-    setVersion(val: number) {
-        this.buffer.writeUInt8(val, BitsUtil.VERSION_FIELD_OFFSET);
+    setVersion(value:number) {
+        this.buffer.writeUInt8(value, BitsUtil.VERSION_FIELD_OFFSET);
     }
 
-    getMessageType(): number {
+    getMessageType():number {
         return this.buffer.readUInt16LE(BitsUtil.TYPE_FIELD_OFFSET);
     }
 
-    setMessageType(val: number) {
-        this.buffer.writeUInt16LE(val, BitsUtil.TYPE_FIELD_OFFSET);
+    setMessageType(value:number) {
+        this.buffer.writeUInt16LE(value, BitsUtil.TYPE_FIELD_OFFSET);
     }
 
-    getFlags(): number {
+    getFlags():number {
         return this.buffer.readUInt8(BitsUtil.FLAGS_FIELD_OFFSET);
     }
 
-    setFlags(val: number) {
-        this.buffer.writeUInt8(val, BitsUtil.FLAGS_FIELD_OFFSET);
+    setFlags(value:number) {
+        this.buffer.writeUInt8(value, BitsUtil.FLAGS_FIELD_OFFSET);
     }
 
-    hasFlags(flags: number): number {
+    hasFlags(flags:number):number {
         return this.getFlags() & flags;
     }
 
-    getFrameLength(): number {
+    getFrameLength():number {
         return this.buffer.readInt32LE(BitsUtil.FRAME_LENGTH_FIELD_OFFSET);
     }
 
-    setFrameLength(val: number) {
-        this.buffer.writeInt32LE(val, BitsUtil.FRAME_LENGTH_FIELD_OFFSET);
+    setFrameLength(value:number) {
+        this.buffer.writeInt32LE(value, BitsUtil.FRAME_LENGTH_FIELD_OFFSET);
     }
 
-    getDataOffset(): number {
+    getDataOffset():number {
         return this.buffer.readInt16LE(BitsUtil.DATA_OFFSET_FIELD_OFFSET);
     }
 
-    setDataOffset(val: number) {
-        this.buffer.writeInt16LE(val, BitsUtil.DATA_OFFSET_FIELD_OFFSET);
+    setDataOffset(value:number) {
+        this.buffer.writeInt16LE(value, BitsUtil.DATA_OFFSET_FIELD_OFFSET);
     }
 
-    getWriteOffset(): number {
-        return this.getDataOffset() + this.writeIndex;
+    appendByte(value:number) {
+        this.buffer.writeUInt8(value, this.cursor);
+        this.cursor += BitsUtil.BYTE_SIZE_IN_BYTES;
     }
 
-    getReadOffset(): number {
-        return this.getDataOffset() + this.readIndex;
+    appendBoolean(value:boolean) {
+        return this.appendByte(value ? 1 : 0);
     }
 
-// PAYLOAD
-
-
-    appendByte(val: number) {
-        this.buffer.writeUInt8(val, this.getWriteOffset());
-        this.writeIndex += BitsUtil.BYTE_SIZE_IN_BYTES;
+    appendInt32(value:number) {
+        this.buffer.writeInt32LE(value, this.cursor);
+        this.cursor += BitsUtil.INT_SIZE_IN_BYTES;
     }
 
-    appendBool(val: boolean) {
-        return this.appendByte(val ? 1 :  0);
+    appendUint8(value:number) {
+        this.buffer.writeUInt8(value, this.cursor);
+        this.cursor += BitsUtil.BYTE_SIZE_IN_BYTES;
     }
 
-    appendInt(val: number) {
-        this.buffer.writeInt32LE(val, this.getWriteOffset());
-        this.writeIndex += BitsUtil.INT_SIZE_IN_BYTES;
-    }
-
-    appendLong(value: any) {
+    appendLong(value:any) {
```

Comment:
I thought about it, but I guess this is a better name since we are using a wrapper for 64-bit integer values, which are not supported natively in javascript.